### PR TITLE
Move MobX assignment type stuff into its own store

### PR DESCRIPTION
**Pull request by pureooze**
Continuing the work from #1043 this PR moves the assignment type stuff into its own store.

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js [hunk 1]
```diff
@@ -136,14 +136,14 @@ class AssignmentTypeEditor extends ActivityEditorMixin(RtlMixin(LocalizeActivity
 	}
 
 	_getInformationText(assignment) {
-		if (!assignment) {
+		if (!assignment || !assignment.assignmentTypeProps) {
```
Comment (kieranderson) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js:139]: Should check `submissionAndCompletionProps` here as well

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-summary.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ class AssignmentTypeSummary extends ActivityEditorMixin(RtlMixin(LocalizeActivit
 
 	render() {
 		const assignment = store.get(this.href);
-		if (assignment && !assignment.isIndividualAssignmentType) {
+		if (assignment && !assignment.assignmentTypeProps.isIndividualAssignmentType) {
```
Comment (kieranderson) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-summary.js:27]: check for `assignmentTypeProps` as well? Similar to the component above,

File: components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment-type.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	setAssignmentTypeGroupCategory(value) {
+		this.selectedGroupCategoryId = value;
+	}
+
+	setToGroupAssignmentType() {
+		this.isIndividualAssignmentType = false;
+		this.selectedGroupCategoryId =
+			this.selectedGroupCategoryId
+				? String(this.selectedGroupCategoryId)
+				: String(this.groupCategories[0].value);
```
Comment (kieranderson) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment-type.js:33]: Is this array guaranteed to have values?

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js (hunk 1 above)
Comment (pureooze) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js:139]: Good catch!

File: components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment-type.js (hunk 3 above)
Comment (pureooze) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment-type.js:33]: Group type seems to be disabled when there are no categories so it should not be possible to call this method when that is the case https://search.d2l.dev/xref/lms/le/dropbox/D2L.LE.Dropbox.Extensibility/HM/Folders/Serializers/Siren/SirenFoldersSerializer.cs?r=ec193248#760

Comment (pureooze) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment-type.js:33]: And siren will set the group type field to disabled in the case that class is present https://github.com/BrightspaceHypermediaComponents/siren-sdk/blob/c4bd360263cd1ae2183c10bdc0fd2c94dd30f73a/src/activities/assignments/AssignmentEntity.js#L249-L258

**Pull request merged**